src/Task/Base/Exec.php

```diff
@@ -132,7 +134,7 @@ public function run()
     {
         $this->hideProgressIndicator();
         // TODO: Symfony 4 requires that we supply the working directory.
-        $result_data = $this->execute(Process::fromShellCommandline($this->getCommand(), getcwd()));
+        $result_data = $this->execute(Process::fromShellCommandline($this->getCommand(), getcwd(), $this->argumentsEnv));
```

Comment:
OK, so this line is using `$argumentsEnv` to set environment variables -- but didn't you already insert the argument values into the command line via `replacePlaceholders`, or is that not in the execution path here?

How do these environment variables interact with `$tastExec->env()`? It looks like there are still test showing that this works.

Reply:
The `replacePlaceholders()` used in `getCommandDescription()` is there to display the command line that will be run by Symfony. The only reason I had to copy that method from SymfonyProcess into ProcessUtils is to rebuild the command line to display it. If there was another way to get a command translated from placeholders to actual value before executing it, it would be preferable, alas Process doesn't seem to provide that feature.
However, it's not meant to rebuild the command line before executing it : let's let Symfony handle that part.

These environment variable are passed as `proc_open()` 5th argument by Process, so it's entirely self-contained into the process we mean to execute, and and won't affect environment variables you can access while in Robo in any means.